File: fisica/js/ejercicio1.js
function hexToRgb(hex) {
    // Verifica si el valor hexadecimal comienza con "#"
    if (hex.startsWith("#")) {
        hex = hex.substring(1);
    }

    // Convierte los dos primeros caracteres hexadecimales a decimal para el componente rojo

    var red = parseInt(hex.substring(0, 2,), 16);
    // Convierte los siguientes dos caracteres hexadecimales a decimal para el componente verde

    var green = parseInt(hex.substring(2, 4), 16);
    // Convierte los últimos dos caracteres hexadecimales a decimal para el componente azul

    var blue = parseInt(hex.substring(4), 16);

    // Retorna un arreglo con los valores de rojo, verde y azul


    return [red, green, blue];
}


var colorHex = "#FF2800";
// Llama a la función hexToRgb() con el valor hexadecimal como argumento

var rgbValues = hexToRgb(colorHex);
console.log("Valores RGB:", rgbValues);

File: fisica/java/ejercicio1.java
import java.util.Arrays;

public class Main {
    public static void main(String[] args) {
        String colorHex = "#FFA500";
        int[] rgbValues = hexToRgb(colorHex);
        System.out.println("Valores RGB: " + Arrays.toString(rgbValues));
    }

    public static int[] hexToRgb(String hex) {
       
        if (hex.startsWith("#")) {
            hex = hex.substring(1);
        }

        
        int red = Integer.parseInt(hex.substring(0, 2), 16);
        int green = Integer.parseInt(hex.substring(2, 4), 16);
        int blue = Integer.parseInt(hex.substring(4), 16);

        return new int[]{red, green, blue};
    }
}
